tests/system/test_deployment_agent_runs.py

```diff
@@ -0,0 +1,175 @@
+    return str(request.param)
+
+
+@pytest.fixture(scope="function")
+def docker_client():
+    client = docker.from_env()
+    yield client
+
+
+# UTILS --------------------------------
+
+
+def get_tasks_summary(tasks):
+    msg = ""
+    for t in tasks:
+        t["Status"].setdefault("Err", "")
+        msg += (
+            "- task ID:{ID}, STATE: {Status[State]}, ERROR: '{Status[Err]}' \n".format(
+                **t
+            )
+        )
+    return msg
+
+
+def get_failed_tasks_logs(service, docker_client):
+    failed_states = ["COMPLETE", "FAILED", "SHUTDOWN", "REJECTED", "ORPHANED", "REMOVE"]
+    failed_logs = ""
+    for t in service.tasks():
+        if t["Status"]["State"].upper() in failed_states:
+            cid = t["Status"]["ContainerStatus"]["ContainerID"]
+            failed_logs += "{2} {0} - {1} BEGIN {2}\n".format(
+                service.name, t["ID"], "=" * 10
+            )
+            if cid:
+                container = docker_client.containers.get(cid)
+                failed_logs += container.logs().decode("utf-8")
+            else:
+                failed_logs += "  log unavailable. container does not exists\n"
+            failed_logs += "{2} {0} - {1} END {2}\n".format(
+                service.name, t["ID"], "=" * 10
+            )
+
+    return failed_logs
+
+
+# TESTS -------------------------------
+
+
+async def test_service_running(service_name, docker_client, loop):
+    """
+    NOTE: Assumes `make up-swarm` executed
+    NOTE: loop fixture makes this test async
+    """
+    running_services = docker_client.services.list()
+    # find the service
+    running_service = [
+        s for s in running_services if service_name == s.name.split("_")[1]
+    ]
+    assert len(running_service) == 1
+
+    running_service = running_service[0]
+
+    # Every service in the fixture runs a single task, but they might have failed!
+    #
+    # $ docker service ps services_storage
+    # ID                  NAME                     IMAGE                     NODE                DESIRED STATE       CURRENT STATE            ERROR                       PORTS
+    # puiaevvmtbs1        services_storage.1       services_storage:latest   crespo-wkstn        Running             Running 18 minutes ago
+    # j5xtlrnn684y         \_ services_storage.1   services_storage:latest   crespo-wkstn        Shutdown            Failed 18 minutes ago    "task: non-zero exit (1)"
+    tasks = running_service.tasks()
+
+    assert (
+        len(tasks) == 1
+    ), "Expected a single task for '{0}'," " got:\n{1}\n{2}".format(
+        service_name,
+        get_tasks_summary(tasks),
+        get_failed_tasks_logs(running_service, docker_client),
+    )
+
+    # wait if running pre-state
+    # https://docs.docker.com/engine/swarm/how-swarm-mode-works/swarm-task-states/
+    pre_states = ["NEW", "PENDING", "ASSIGNED", "PREPARING", "STARTING"]
+
+    for n in range(RETRY_COUNT):
+        task = running_service.tasks()[0]
```

Comment:
using task[0] is actually not correct since it's not always the latest task. 
instead you could use something like
```python
tasks_current_state = [task["Status"]["State"] for task in tasks]
num_running = sum(current == "running" for current in tasks_current_state)

```

Reply:
It is fine (not the prettiest code but anyway, fine for this PR), because before this call we assert that only one task is running, ie len(tasks) is 1